CMakeLists.txt

```diff
@@ -348,7 +349,10 @@ add_definitions(-DBOOST_TEST_DYN_LINK)
 #   ... openMP code here ...
 # }
 # #endif
-find_package(OpenMP)
+if (USE_OPENMP)
+  find_package(OpenMP)
+endif ()
+
 if (OPENMP_FOUND)
```

Comment:
I'm wondering what's ``OPENMP_FOUND`` if ``USE_OPENMP`` is ``OFF``; undefined? So there is no need to do something like?:

```
if (USE_OPENMP  AND OPENMP_FOUND)
...
```

Reply:
`OPENMP_FOUND` should be an unset variable, and CMake's `if()` operator will evaluate to false when a `if(<variable>)` is called for an undefined variable:

https://cmake.org/cmake/help/v3.0/command/if.html